libp2p-websocket/src/browser.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a 
+// copy of this software and associated documentation files (the "Software"), 
+// to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+// and/or sell copies of the Software, and to permit persons to whom the 
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in 
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS 
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+// DEALINGS IN THE SOFTWARE.
+
+use std::io::{Read, Write};
+use std::io::{Error as IoError, ErrorKind as IoErrorKind};
+use std::sync::{Arc, Mutex};
+use futures::{Future, Stream, Poll, Async, Then as FutureThen};
+use futures::sync::{oneshot, mpsc};
+use futures::stream::Then as StreamThen;
+use multiaddr::{Multiaddr, AddrComponent};
+use rw_stream_sink::RwStreamSink;
+use stdweb::{self, Reference};
+use stdweb::web::TypedArray;
+use swarm::Transport;
+use tokio_io::{AsyncRead, AsyncWrite};
+
+/// Represents the configuration for a websocket transport capability for libp2p.
+///
+/// This implementation of `Transport` accepts any address that looks like
+/// `/ip4/.../tcp/.../ws` or `/ip6/.../tcp/.../ws`, and connect to the corresponding IP and port.
+///
+/// > **Note**: The `/wss` protocol isn't supported.
+#[derive(Debug, Clone)]
+pub struct WsConfig;
+
+impl WsConfig {
+    /// Creates a new configuration object for websocket.
+    #[inline]
+    pub fn new() -> WsConfig {
```

Comment:
I don't like how the signature for this method changes depending on your platform but otherwise everything acts the same, it seems like it's just asking for confusing error messages when you switch platform and expect everything to work the same. Maybe have `BrowserWsConfig` and `WsConfig`?

Reply:
It is also true that `WsConfig` (the desktop one) could be used from inside nodejs or another similar environment, which makes the name `BrowserWsConfig` more relevant.